doca-dpdk: update the pipeline entry remove handle

jiaweiwsz opened the pull request:
Update the pipeline entry remove function, firstly find
the matched entry based on the pipe_queue, if found, then
free entry and update the numbers of total pipe entries.

Adds the input argument checking and return if NULL
pointer;

Signed-off-by: Jiawei Wang <jiaweiw@nvidia.com>

jiaweiwsz commented:
@rbaryanai , Please help to review it code change for doca flow API error handle and update the doca_flow_pipe_rm_entry function implementation.

Thanks.

rbaryanai commented on examples/vnet/doca_dpdk.c:
```diff
@@ -1496,6 +1509,22 @@ doca_dpdk_add_pipe_entry(struct doca_flow_pipe *pipe,
+	struct doca_flow_pipe_entry *entry;
+	int found = 0;
+
+	LIST_FOREACH(entry, &pipe->entry_list[pipe_queue], next) {
```
we should never use linked-list for more than few objects.
see my comment below, you don't need to find it.

jiaweiwsz replied:
OK, 
I have one question how to use 'pipe_queue' in doca_flow_pipe_rm_entry() if not need search?
Only need the entry is enough to remove the resource.

rbaryanai commented on examples/vnet/doca_flow.c:
```diff
@@ -64,10 +63,26 @@ int
+	if (entry == NULL || entry->flow_pipe == NULL)
+		return -EINVAL;
+	pipe = (struct doca_flow_pipe *)entry->flow_pipe;
+	if (doca_dpdk_find_pipe_entry(pipe_queue, pipe, entry)) {
```
if entry list pointers are not NULL or trahsed.

jiaweiwsz replied:
Done.

jiaweiwsz committed:
doca-dpdk: update the pipeline entry remove handle

Updates the pipeline entry remove function to free the entry
resources, flows belong to this entry and entry self, also
updates the numbers of total pipe entries.

Adds the input argument checking for NULL pointer.

Signed-off-by: Jiawei Wang <jiaweiw@nvidia.com>
